ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/Topic.java

```diff
@@ -58,11 +87,39 @@ public SerdeSupplier getSerdeSupplier() {
     return serdeSupplier;
   }
 
+  public SerdeFactory getKeySerdeFactory() {
+    return keySerdeFactory;
+  }
+
   Serializer getSerializer(final SchemaRegistryClient schemaRegistryClient) {
     return serdeSupplier.getSerializer(schemaRegistryClient);
   }
 
   Deserializer getDeserializer(final SchemaRegistryClient schemaRegistryClient) {
     return serdeSupplier.getDeserializer(schemaRegistryClient);
   }
+
+  Serializer getKeySerializer() {
+    return keySerdeFactory.create().serializer();
+  }
+
+  @SuppressWarnings("unchecked")
+  Deserializer getKeyDeserializer() {
+    final Serde keySerde = keySerdeFactory.create();
+    if (keySerde instanceof TimeWindowedSerde) {
+      final TimeWindowedSerde windowedSerde = (TimeWindowedSerde) keySerde;
```

Comment:
What's the purpose of this logic here (and of having `Topic` store a `windowSize`)? Is it not possible to create the Topic with the correct key serde from the get-go and avoid the `Topic` needing to store a window size?

Reply:
That was the initial approach but since the serde in KSQL does not store the window size we need to keep track of it ourselves.
We can add the window size to the serde object in KSQL and that would eliminate the need to keep track of it here.